### PR TITLE
fix: enable enrollment for courses with past upgrade deadline

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#4975 

#### What's this PR do?
- Enabled the enrollment for courses that have an upgrade deadline I the past
- fixes the display for `View on edX` and enrollment buttons

#### How should this be manually tested?
Steps from #4975 can be followed to test it

#### Screenshots (if appropriate) *(Updated)
**On dashboard( FA program -- When the user has not added/calculated a request for personal price and upgrade deadline for course is in past)**
Screenshot 2021-07-26 at 1 53 24 PM

**On dashboard( FA program -- When the user has added/calculated a request for personal price but its in pending state, and upgrade deadline for course is in past)**
Screenshot 2021-07-26 at 1 56 28 PM
**On dashboard( FA program -- When the user personal price is approved and upgrade deadline for course is in past)**

Screenshot 2021-07-26 at 1 57 32 PM

**On dashboard( Non-FA program)**
Screenshot 2021-07-26 at 1 58 57 PM

Screenshot 2021-07-16 at 5 54 44 PM

**On program page**
Screenshot 2021-07-16 at 5 48 23 PM

### CODE REVIEW

**Review by annagav — CHANGES_REQUESTED** [review]
I think we should remove the option to pay in the dialog. Or disable it and add a message saying the the deadline to upgrade has passed, so they can't pay.

**Comment by annagav** [comment]
Also, the are two messages are a somewhat confusing. The course progress message says "course stared" and the course status message says "course is starting"

**Comment by arslanashraf7** [comment]
> I think we should remove the option to pay in the dialog. Or disable it and add a message saying the the deadline to upgrade has passed, so they can't pay.

@annagav I've updated the PR with the below changes, let me know your thoughts:
1. Updated course status/progress messages in case of the future enrollment end date and past course start date.
2. Disabled the `Pay Now` button based on `Upgrade deadline`, if the `course upgrade deadline` is in past then the `Pay Now` button will be disabled for all the users either they are new or they come to upgrade the course due to other reason.

If the above perception is okay with you, I'll mark the PR ready for review.

**Comment by annagav** [comment]
@pdpinch @briangrossman could you confirm that the copy changes are okay.

**Comment by annagav** [comment]
@pdpinch @briangrossman Could you confirm the new dialog message "You can't pay for this course. The deadline to upgrade the course has passed."

File: static/js/components/CourseEnrollmentDialog.js [hunk 1]
```diff
@@ -121,6 +122,25 @@ export default class CourseEnrollmentDialog extends React.Component {
...
         </button>
       )
+    } else if (
+      courseRun.course_upgrade_deadline &&
+      moment(courseRun.course_upgrade_deadline).isBefore(moment(), "day")
+    ) {
+      message = (
+        <p key="1">
+          You can't pay for this course, The deadline to upgrade the course has
+          passed.
```
Comment (pdpinch) [inline static/js/components/CourseEnrollmentDialog.js:132]:
```suggestion
          You cannot pay for this course at this time. The payment deadline has passed.
```

**Comment by pdpinch** [comment]
Will this message apply to all programs, i.e. both the fa program (DEDP, with it's proctored exam system) and the non-fa programs (e.g. SCM) ?

Can you update the screenshot?

**Comment by arslanashraf7** [comment]
> Will this message apply to all programs, i.e. both the fa program (DEDP, with it's proctored exam system) and the non-fa programs (e.g. SCM) ?

Yes, it will be applied to both FA and Non-FA programs. However, the order of messages shown based on the conditions for FA programs can be seen in the screenshot descriptions.

Code reference for the order of messages for FA is listed below in orderly fashion from top-bottom:
- [Personal price is in pending state](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L89)
- [User has already paid](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L109)
- [Upgrade deadline has passed](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L125)
- [Personal price is approved/User had applied](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L144)
- [User has not calculated the personal price](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L165)
 
> Can you update the screenshot?

Updated

**Review by annagav — APPROVED** [review]
looks good 👍

**Comment by annagav** [comment]
@umarmughal824 I am sorry for the misunderstanding, but Peter and Brian have to confirm your text changes.

**Comment by briangrossman** [comment]
👍 
I'm fine with this as is.

I'm going to create a new issue with some minor text tweaks, but this PR can be released.

**Pull request merged**